### ISSUE TITLE
[a11y]: Keyboard focus is trapped in FileUploader

### ISSUE THREAD

**Issue by kzhang-2017** [carbon-design-system/carbon#15870]
### Package

carbon-components, carbon-components-react

### Browser

Chrome

### Operating System

MacOS

### Package version

v10.58.10

### React version

v16.14.0

### Automated testing tool and ruleset

Manual testing

### Description

Expected to be able to tab out of the FileUploader button. Cannot tab out once you're in the button. 

Can reproduce in carbon 10 but not carbon 11. Can we backport the fix there as well?

### WCAG 2.1 Violation

No Keyboard Trap

### Reproduction/example

import { FileUploader } from '@carbon/react'; import React from 'react';  export default function App() {   return (     <div>       <FileUploader buttonLabel={t('upload')} />     </div>   ); }

### Steps to reproduce

1. Hit tab until you are inside the `FileUploader` component. 
2. Cannot tab out once you are in that button

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
